config/token.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"github.com/dgrijalva/jwt-go"
+	"time"
+)
+
+// Structured version of Claims Section, as referenced at
+// https://tools.ietf.org/html/rfc7519#section-4.1
+// See examples for how to use this with your own claim types
+type TokenClaim struct {
+	User string `json:"username"`
+	jwt.StandardClaims
+}
+
+type TokenGenerated struct {
+	Token     string `json:"token"`
+	ExpiredAt string `json:"expired_at"`
+}
+
+/*
+Generate the token with a Expiraton of <ExpiresAt> seconds
+*/
+func GenerateToken(username string) (TokenGenerated, error) {
+	claim := &jwt.StandardClaims{}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claim)
+	ss, err := token.SignedString(Get().Token.Secret)
+	if err != nil {
+		return TokenGenerated{}, err
+	}
+
+	return TokenGenerated{Token: ss, ExpiredAt: time.Now().Add(time.Second * time.Duration(Get().Token.ExpirationAt)).String()}, nil
+}
+
+func ValidateToken(tokenString string) error {
+	token, err := jwt.ParseWithClaims(tokenString, &jwt.StandardClaims{}, func(token *jwt.Token) (interface{}, error) {
+		return Get().Token.Secret, nil
+	})
+	if err != nil {
+		fmt.Println("Here")
+		return err
+	}
+	if token.Valid {
```

Comment:
I think you are missing an step. Validate the "alg" claim. (taken from [here](https://godoc.org/github.com/dgrijalva/jwt-go#example-Parse--Hmac))

```golang
// Don't forget to validate the alg is what you expect:
    if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
        return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
    }
```

If not validated, one could create a JWT with the "alg = none" and bypass your protection. 
alg none means that the jwt is unsecured, but is still a valid JWT, and the RFC requires [1] that the libs implement that one.

See this post [2] for more info.
> The none algorithm is a curious addition to JWT. It is intended to be used for situations where the integrity of the token has already been verified. Interestingly enough, it is one of only two algorithms that are mandatory to implement (the other being HS256).


[1] https://tools.ietf.org/html/rfc7519#section-8
[2]https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/

Reply:
We could also configure [validMethods](https://github.com/dgrijalva/jwt-go/blob/master/parser.go#L30)

Reply:
Well, it seems that `none` is disabled by default https://github.com/dgrijalva/jwt-go/blob/master/none.go#L28
But anyway, we should only allow tokens signed with `SigningMethodHS256`